src/include/server.h

```diff
@@ -253,6 +253,9 @@ enum failover_state {
 #define SVR_CLEAN_JOBHIST_SECS	5	/* never spend more than 5 seconds in one sweep to clean hist */
 #define SVR_JOBHIST_DEFAULT	1209600	/* default time period to keep job history: 2 weeks */
 
+#define VALUE(str) #str
+#define TOSTR(str) VALUE(str)
```

Comment:
Well, isn't that clever.  :-)

Reply:
I think I'm missing something here.  Why do you need two macros?  Why can't you just have one macro that does #str?  According to my book, the # preprocessor will quote what follows.  I don't see what having the extra macro buys us.

Reply:
I would like share you how I came to write such a macro. 

Following is the simple program that I tried initially.

#include <stdio.h>
#define PBS_SCHEDULE_CYCLE    600 /* re-schedule even if no change, 10 min   */

#define VALUE(str) #str
#define TOSTR(str) VALUE(str)

unsigned char s1[60] = TOSTR (PBS_SCHEDULE_CYCLE);    /* generates "600" \*/
unsigned char s2[60] = VALUE(PBS_SCHEDULE_CYCLE);     /* generates "PBS_SCHEDULE_CYCLE" \*/
int main()
{
        printf("s1 = %s\n",s1);
        printf("s2 = %s\n",s2);
        return 0;
}
Since we need the value of PBS_SCHEDULE_CYCLE as a string, I had to use both of the above macros.

Reply:
FWIW, it looks correct to me.

Reply:
In my opinion, I would flip them. `TOSTR` should return the name of the `#define` as a string, and `VALUE` should return the value of `#define`.

Reply:
Thanks for the explanation.  I see it now.  I agree with @mike0042, this is quite clever.  As to @vstumpf's comment, the usable macro is TOSTR().  The other macro can be called anything.  I think TOSTR() correctly defines what is going on.

Reply:
Ah, I see now that TOSTR is making the value of a `#define` to a string. Makes sense to me.